appinventor/appengine/src/com/google/appinventor/client/widgets/SearchBox.java

```diff
@@ -0,0 +1,203 @@
+                }
+            }
+        }) ;
+
+        searchBox.addKeyPressHandler(new KeyPressHandler() {
+            @Override
+            public void onKeyPress(KeyPressEvent event) {
+                String queried = searchBox.getText();
+                int keyCode = event.getCharCode();
+                try {
+                  MultiWordSuggestOracle multiWordOracle = (MultiWordSuggestOracle)searchBox.getSuggestOracle();
+                  List<String> suggestions = blockArea.querySuggest(queried);
+                  for(String suggestion:suggestions){
+                    multiWordOracle.add(suggestion);
+                  }
+                } catch (ClassCastException e ) {
+
+                }
+            }
+                  
+        }) ;
+
+        /*Clear Button*/
+        clearButton = new TextButton("X");
+        clearButton.setHeight("20px");
+        clearButton.addClickHandler(new ClickHandler() {
+            public void onClick(ClickEvent click) {
+                searchBox.setValue("");
+                blockArea.stopSearch();
+            }
+        });
+
+
+        /*Filter Box */
+        filterBox = new ListBox();
+        filterBox.getElement().getStyle().setProperty("color", "grey");
+        filterBox.addItem("Please select one....");
+        filterBox.setHeight("20px");
+        flag = 0;
+        filterBox.addClickHandler(new ClickHandler() {
+            @Override
+            public void onClick(ClickEvent event) {
+                if(flag == 0){
+                    try{    
+                            flag = 1;
+                            filterBox.clear();
+                            List<String> types = blockArea.getBlockTypes("");
+                            filterBox.addItem("Please select one....");
+                            for(String type:types){
+                               filterBox.addItem(type);
+                            }
+                        } catch (ClassCastException e ) {
+
+                        }
+                    }
+              }
+            });
+        
+
+
+        /* Filters Check Box */
+        advancedCheckBox = new CheckBox("Filters");
+        advancedCheckBox.getElement().getStyle().setProperty("color", "grey");
+        advancedCheckBox.setHeight("20px");
+        advancedCheckBox.addClickHandler(new ClickHandler() {
+          @Override
+          public void onClick(ClickEvent event) {
+            boolean checked = ((CheckBox) event.getSource()).getValue();
+            if(checked){
+                filterBox.getElement().getStyle().setProperty("color", "black");
+                try {
+                  filterBox.clear();
+                  filterBox.addItem("Please select one....");
+                  List<String> types = blockArea.getBlockTypes("");
+                  for(String type:types){
+                    filterBox.addItem(type);
+                  }
+                } catch (ClassCastException e ) {
+
```

Comment:
what happens here? If an exception is safe to ignore, add a comment that states why. If not, please handle it.